packages/sommelier.rb

```diff
@@ -2,106 +2,266 @@
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> initsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> initsommelier"
-      system "echo 'else' >> initsommelier"
-      system "echo '  echo \"sommelier failed to start\"' >> initsommelier"
-      system "echo '  exit 1' >> initsommelier"
-      system "echo 'fi' >> initsommelier"
-      system "echo '#!/bin/bash' > stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  killall -g sommelier' >> stopsommelier"
-      system "echo '  if pgrep \"Xwayland\"; then killall Xwayland; fi'"
-      system "echo '  sleep 3' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  echo \"sommelier process stopped\"' >> stopsommelier"
-      system "echo 'else' >> stopsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> stopsommelier"
-      system "echo '  exit 1' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
-      system "echo '#!/bin/bash' > startsommelier"
-      system "echo 'set -a && source ~/.sommelier.env && set +a && initsommelier' >> startsommelier"
-    end
+  case ARCH
+   when 'armv7l', 'aarch64'
+      PEER_CMD_PREFIX='/lib/ld-linux-armhf.so.3'
+   when 'i686'
+      PEER_CMD_PREFIX='/lib/ld-linux-i686.so.2'
+   when 'x86_64'
+      PEER_CMD_PREFIX='/lib64/ld-linux-x86-64.so.2'
   end
+    
+  def self.build
+    # There is no good way to checksum the googlesource tgz file, as they appear to be generated on the fly
+    # and checksums vary with each download.
+    system 'curl -L https://chromium.googlesource.com/chromiumos/platform2/+archive/c6f85a3472584df37fcf2d3d99474081f29256cd.tar.gz | tar mzx --warning=no-timestamp'
+    Dir.chdir ("vm_tools/sommelier") do
+    
+    # Google's sommelier expects to find virtwl.h in their kernel source includes, but we may not have
+    # set of kernel headers which match, so we just download virtwl.h and then patch the sommelier source
+    # to look for the file locally.
+    ######################### Download virtwl.h from Chromium 5.4 kernel tree ###########################################
+    url_virtwl = "https://chromium.googlesource.com/chromiumos/third_party/kernel/+/5d641a7b7b64664230d2fd2aa1e74dd792b8b7bf/include/uapi/linux/virtwl.h?format=TEXT"
+    uri_virtwl = URI.parse url_virtwl
+    #filename_virtwl = File.basename(uri_virtwl.path)
+    filename_virtwl = 'virtwl.h_base64'
+    sha256sum_virtwl = 'a8215f4946ccf30cbd61fcf2ecc4edfe6d05bffeee0bacadd910455274955446'
+    
+    if File.exist?(filename_virtwl) && Digest::SHA256.hexdigest( File.read( filename_virtwl ) ) == sha256sum_virtwl
+      puts "Unpacking virtwl source code".yellow
```

Comment:
Still a bit foggy on this logic.  What are we "unpacking" here?  How will this file ever exist until it is downloaded via curl first?

Reply:
This is due to the logic being cribbed from another package file that was downloading patches, IIRC.

Reply:
Can we remove this conditional like the other package?

Reply:
Done.